src/main/java/ch/njol/skript/effects/EffSendBlockChange.java

```diff
@@ -34,84 +32,80 @@
-@Since("2.2-dev37c, 2.5.1 (block data support)")
+@Description("Makes players see blocks as something else.")
+@Examples({
+	"make player see block at player as dirt",
+	"make player see target block as campfire[facing=south]"
+})
+@Since("2.2-dev37c, 2.5.1 (block data support), INSERT VERSION (reset)")
 public class EffSendBlockChange extends Effect {
 
-	private static final boolean BLOCK_DATA_SUPPORT = Skript.classExists("org.bukkit.block.data.BlockData");
-	private static final boolean SUPPORTED =
-			Skript.methodExists(
-					Player.class,
-					"sendBlockChange",
-					Location.class,
-					Material.class,
-					byte.class
-			);
-
 	static {
 		Skript.registerEffect(EffSendBlockChange.class,
-				BLOCK_DATA_SUPPORT ? "make %players% see %blocks% as %itemtype/blockdata%" : "make %players% see %blocks% as %itemtype%"
-		);
+				"make %players% see %blocks% as %itemtype/blockdata%",
+				"make %players% see %blocks% (as normal|the same as the server)");
 	}
 
-	@SuppressWarnings("null")
-	private Expression<Player> players;
-
-	@SuppressWarnings("null")
+	@Nullable
+	private Expression<Object> as;
 	private Expression<Block> blocks;
+	private Expression<Player> players;
 
-	@SuppressWarnings("null")
-	private Expression<Object> as;
-	
 	@Override
 	@SuppressWarnings("unchecked")
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
-		if (!SUPPORTED) {
-			Skript.error("The send block change effect is not supported on this version. " +
-				"If Spigot has added a replacement method without magic values " +
-				"please open an issue at https://github.com/SkriptLang/Skript/issues " +
-				"and support will be added for it.");
-			return false;
-		}
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		players = (Expression<Player>) exprs[0];
 		blocks = (Expression<Block>) exprs[1];
-		as = (Expression<Object>) exprs[2];
+		if (matchedPattern == 0)
+			as = (Expression<Object>) exprs[2];
 		return true;
 	}
 
 	@Override
-	protected void execute(Event e) {
-		Object object = this.as.getSingle(e);
-		if (object instanceof ItemType) {
-			ItemType itemType = (ItemType) object;
-			for (Player player : players.getArray(e)) {
-				for (Block block : blocks.getArray(e)) {
+	protected void execute(Event event) {
+		Object object = null;
+		if (this.as != null) {
+			object = this.as.getSingle(event);
+			if (object == null)
+				return;
+		}
+		// Reset the block the players are seeing to be synced with server.
+		if (object == null) {
+			for (Player player : players.getArray(event)) {
+				for (Block block : blocks.getArray(event)) {
+					player.sendBlockChange(block.getLocation(), block.getBlockData());
+				}
+			}
+		} else if (object instanceof ItemType itemType) {
+			for (Player player : players.getArray(event)) {
+				for (Block block : blocks.getArray(event)) {
```

Comment:
this evaluates the blocks expression multiple times
getArray needs to be done outside the loop